File: backend/src/main/kotlin/reversi/model/Game.kt
package reversi.model

import kotlinx.serialization.Serializable
import reversi.controller.dto.MoveRequest
import java.util.*

@Serializable
data class Game(
    val id: String = UUID.randomUUID().toString(),
    val difficulty: Difficulty = Difficulty.MEDIUM,
    val board: GameBoard,
    val currentPlayer: CellState = CellState.BLACK,
    val validMoves: List<MoveRequest> = emptyList(),
    val playerTypes: Map<CellState, PlayerType> = mapOf(
        CellState.BLACK to PlayerType.HUMAN,
        CellState.WHITE to PlayerType.AI
    ),
    val isFinished: Boolean = false
)

enum class PlayerType { HUMAN, AI }
enum class Difficulty { EASY, MEDIUM, HARD }

File: frontend/src/types/game.ts
export type PlayerType = 'HUMAN' | 'AI'
export type Player = 'BLACK' | 'WHITE'
export type CellState = Player | 'EMPTY'

export interface Move {
  row: number
  col: number
}

export interface GameState {
  id: string
  currentPlayer: Player
  isFinished: boolean
  board: { grid: CellState[][] }
  validMoves: Move[]
}

export interface PlayerTypes {
  BLACK: PlayerType
  WHITE: PlayerType
}

export interface NewGameRequest {
  id?: string
  difficulty: 'EASY' | 'MEDIUM' | 'HARD'
  playerTypes: PlayerTypes
  currentPlayer: Player
  preferredSide: Player
}
